src/react-app-env.d.ts

```diff
@@ -1 +1,23 @@
 /// <reference types="react-scripts" />
+
+interface User {
+  name: string;
+  email: string;
+}
+
+interface CommentItem {
+  postId: number,
+  id: number,
+  name: string,
+  email: string,
+  body: string,
+}
+
+interface Post {
+  userId: number;
+  id: number;
+  title: string;
+  body: string;
+  user: User | null;
+  comments: CommentItem[];
+}
```

Comment:
You can also inherit some instance and use union interface where need it
`interface Post {
  userId: number;
  id: number;
  title: string;
  body: string;
}
interface prepearedPost extend Post {
  user: User | null;
  comments: CommentItem[];
}
`